ipaserver/install/ipa_server_certinstall.py

```diff
@@ -147,21 +149,35 @@ def install_dirsrv_cert(self):
                                ['nssslpersonalityssl'])
         old_cert = entry.single_value['nssslpersonalityssl']
 
-        server_cert = self.import_cert(dirname, self.options.pin,
-                                       old_cert, 'ldap/%s' % api.env.host,
-                                       'restart_dirsrv %s' % serverid)
+        nickname, cert = self.import_cert(dirname, self.options.pin,
+                                          old_cert, 'ldap/%s' % api.env.host,
+                                          'restart_dirsrv %s' % serverid)
 
-        entry['nssslpersonalityssl'] = [server_cert]
+        entry['nssslpersonalityssl'] = [nickname]
         try:
             conn.update_entry(entry)
         except errors.EmptyModlist:
             pass
 
+        ds = dsinstance.DsInstance()
+        ds.suffix = api.env.basedn
+        ds.realm = api.env.realm
+        ds.fqdn = api.env.host
+        ds.service_prefix = 'ldap'
+        ds.cert = cert
+        name = f'{ds.service_prefix}/{ds.fqdn}'
+        oldcerts = api.Command.cert_find(service=name)['result']
+        ds.add_cert_to_service(append=False)
```

Comment:
The option append=False introduces a slightly different behavior for IPA-issued certs and 3rd-party certs.
When the LDAP server cert is issued by IPA and renewed by IPA, the new certificate is added to the service entry and the old one is kept: https://github.com/freeipa/freeipa/blob/master/ipaserver/plugins/cert.py#L961-L968
With this code, if the LDAP server cert is a 3rd-party cert, it replaces the old cert in the service entry (the old cert is removed).

IMO the logic should be similar in both cases (either always replace or always add). No strong opinion on the best choice.

Reply:
It's been a while so I forget my reasoning. I think it may be related to what cert-find returned since it also looks in the service cert values. I'll double-check.